### ISSUE TITLE
non-breaking space in match case leads to strange error message

### ISSUE THREAD

**Issue by lambdafu** [coq/coq#17494]
#### Description of the problem

```
Definition func (b1: bool) (b2:bool) : bool :=
match b1 with
| true => true
| false => false
end.
```

results in

```
Pattern "false" is redundant in this clause.
```

if there is a non-breaking space between "|" and "true" in the third line. I entered this by accident on a MacBook in the coqide (with option-space), and had to use a hexdump tool to find the problem. It was very confusing!

#### Coq Version
8.16 22.09, macOS M1

**Comment by lambdafu** [comment]
Might be related to #10168

**Comment by Zimmi48** [comment]
I was also hit by this issue two days ago. I had a student who accidentally included a non-breaking space in a constructor in an inductive type definition, resulting in the same error when writing a match. This was very confusing and difficult to debug.

**Comment by ppedrot** [comment]
This is not specific to pattern-matching, and happens all the time on my configuration (here AltGr + space results in a NBSP). You always get confusing errors with this. I think that we should display such space-looking characters specially in the output.

**Comment by SkySkimmer** [comment]
>I think that we should display such space-looking characters specially in the output.

I don't think this would help in this case, the nbsp is not present in the output.

**Comment by SkySkimmer** [comment]
We already treat nbsp specially https://github.com/coq/coq/blob/03e9548c9d880205cb71ca36252ce9aeef5c3452/clib/unicode.ml#L134
Maybe it should be identpart instead of identsep (not allowed at the start of ident)? Or some other category?
